internal/pkg/metadb/metadb.go

```diff
@@ -524,3 +527,49 @@ func (m *MetaDB) ListBlobRefsByStatus(ctx context.Context, status blobref.Status
 	iter := blobref.NewCursor(m.client.Run(ctx, query))
 	return iter, nil
 }
+
+func addPropertyFilter(q *ds.Query, f *pb.QueryFilter) *ds.Query {
+	switch f.Operator {
+	case pb.FilterOperator_EQUAL:
+		return q.Filter(fmt.Sprintf("Properties.%s =", f.PropertyName), record.ExtractValue(f.Value))
+	default:
+		// TODO(hongalex): implement inequality filters
+	}
+	return q
+}
+
+// QueryRecords returns a list of records that match the given filters and their stores.
+func (m *MetaDB) QueryRecords(ctx context.Context, filters []*pb.QueryFilter, storeKey, owner string, tags []string) ([]*pb.Record, []string, error) {
```

Comment:
Can we use the record.Record type instead of the protobuf here just like other functions in the package.

Reply:
This might be a premature optimization, but I can call `record.ToProto()` as I construct the slice here. If I pass `record.Record` as the slice type, I'll have to iterate over the array, calling `ToProto` again in `QueryRecords` in `open_saves.go`. Is this tradeoff worth it?

Reply:
That's a good point. I think we'd still call ToProto() once for each record and iterating over slices is not very expensive, so I'd opt to consistency here. What do you think?

Reply:
Made the change for consistency